### PR TITLE
Read [[LastCreatedOffer/Answer]] at the right time in SLD

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes https://github.com/w3c/webrtc-pc/issues/2208.

***
<a href="https://pr-preview.s3.amazonaws.com/jan-ivar/webrtc-pc/pull/2209.html" title="Last updated on Jul 11, 2019, 1:16 PM UTC (e326fc9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2209/3e20f80...jan-ivar:e326fc9.html" title="Last updated on Jul 11, 2019, 1:16 PM UTC (e326fc9)">Diff</a>

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -3023,27 +3021,48 @@ <h2>Methods</h2>
...
+                        then set <var>description.sdp</var> to the value of
+                        <var>connection</var>'s <a>[[\LastCreatedOffer]]</a>
+                        slot.</p>
+                      </li>
+                      <li>
+                        <p>Return the result of <a
+                        href="#set-description">setting the RTCSessionDescription</a>
+                        indicated by <code><var>description</var></code>.</p>
+                      </li>
+                    <ol>
```
Comment (dontcallmedom) [inline webrtc.html]: this should be `</ol>` (hence the Travis CI error on this and #2212)

File: webrtc.html [hunk 2]
```diff
@@ -3023,27 +3021,48 @@ <h2>Methods</h2>
...
+                  <li>
+                    <p>Let <var>connection</var> be the
+                    <code><a>RTCPeerConnection</a></code> object on which the
+                    method was invoked.</p>
+                  </li>
+                  <li>
+                    <p>Return the result of <a href=
+                    "#enqueue-an-operation">enqueuing</a> the following steps to
+                    <var>connection</var>'s operation queue:</p>
+                    <ol>
```
Comment (henbos) [inline webrtc.html]: We are enqueuing a task that sets the session description (= creates a promise and then enqueues a task to resolve it). It doesn't make sense to me to enqueue a task to enqueue a task.
1. It adds an additional delay, that sounds like a bug to me. While it probably rarely creates problems, it doesn't seem to add any value to do so.
2. Language-wise, it doesn't make sense to return the result of enqueuing an operation. The promise we want to return hasn't been created at this point. We would have to create another promise and wrap the other promise in it. But see 1), we don't want to do that.

File: webrtc.html [hunk 3]
```diff
@@ -3059,10 +3078,29 @@ <h2>Methods</h2>
...
+                  </li>
+                  <li>
+                    <p>Let <var>connection</var> be the
+                    <code><a>RTCPeerConnection</a></code> object on which the
+                    method was invoked.</p>
+                  </li>
+                  <li>
+                    <p>Return the result of <a href=
+                    "#enqueue-an-operation">enqueuing</a> the following steps
+                    to <var>connection</var>'s operation queue:</p>
```
Comment (henbos) [inline webrtc.html]: Same as above: Why are we enqueuing a task to enqueue a task? We should just say to return the result of setting the RTCSessionDescription immediately.

File: webrtc.html [hunk 4]
```diff
@@ -3087,6 +3125,7 @@ <h2>Methods</h2>
                 <code>setRemoteDescription</code> does not await the completion
                 of identity validation.</p>
               </dd>
+
```
Comment (henbos) [inline webrtc.html]: Accidental newline?

**Comment by henbos** [comment]
> I think we moved the problem instead of fixing it

Correction: The race of the referenced bug is fixed here, but still enqueuing at a later stage is probably unnecessary and possibly problematic (correct me if I'm wrong)

File: webrtc.html (hunk 2 above)
Comment (jan-ivar) [reply webrtc.html]: Resolved offline. [Enqueueing an operation](http://w3c.github.io/webrtc-pc/#enqueue-an-operation) is not the same as [queue a task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-a-task). The former chains an asynchronous operation on the `RTCPeerConnection`'s internal serialization queue. The latter takes us back to main thread, adding an event to the JS event loop.

Comment (jan-ivar) [reply webrtc.html]: The former also returns a promise, so we should be good here.

**Pull request merged**